src/components/BounceRuleDetailed/Changelog/index.jsx

```diff
@@ -7,45 +7,85 @@ import {
   TableHeader,
   TableRow,
 } from "@sendgrid/ui-components/table/table";
-import mockChangelog from "../../../mocks/index";
+import Moment from "react-moment";
+import Card from "@sendgrid/ui-components/card";
+import { CSVLink } from "react-csv";
+import { Row } from "../../Row";
+import { Column } from "../../Column";
 
-const Changelog = ({ handleButtonClicked }) => (
-  <div>
-    <h2>Changelog</h2>
-    <Table>
-      <TableHeader>
-        <TableRow>
-          <HeaderCell>Date</HeaderCell>
-          <HeaderCell>User</HeaderCell>
-          <HeaderCell>Commit Message</HeaderCell>
-          <HeaderCell className="actions-align">Actions</HeaderCell>
-        </TableRow>
-      </TableHeader>
-      <TableBody>
-        {mockChangelog.map(change => (
-          <ChangelogMin
-            key={change.id}
-            change={change}
-            handleButtonClicked={handleButtonClicked}
-          />
-        ))}
-      </TableBody>
-    </Table>
+import { WriteSelectors } from "../selectors";
+
+const EmptyChangelog = () => (
+  <Row>
+    <Card
+      icon="airplane"
+      iconSize={40}
+      title="No Previous Changes Found!"
+      body="No changes have been previously made to this bounce rule."
+      centered
+    />
+  </Row>
+);
+
+const Changelog = ({
+  handleChangelogClicked,
+  changelog,
+  isChangelogEmpty,
+  changelogLimit,
+}) => (
+  <div {...WriteSelectors.changelog}>
+    <Row id="changelog-title">
+      <Column width={1} offset={1}>
+        <h2>Changelog</h2>
+      </Column>
+      <Column width={2} offset={11} className="changelog-csv sg-right">
+        <CSVLink data={changelog} filename="changelog.csv" target="_blank">
+          Export as CSV
+        </CSVLink>
+      </Column>
+    </Row>
+    {isChangelogEmpty && <EmptyChangelog />}
+    {!isChangelogEmpty && (
+      <Table>
+        <TableHeader>
+          <TableRow>
+            <HeaderCell>Date</HeaderCell>
+            <HeaderCell>User</HeaderCell>
+            <HeaderCell>Commit Message</HeaderCell>
+            <HeaderCell className="actions-align">Actions</HeaderCell>
+          </TableRow>
+        </TableHeader>
+        <TableBody>
+          {changelog.slice(0, changelogLimit).map((change, index) => (
```

Comment:
We can encapsulate all this in a function like {this.showChangelogMin()} so it's a bit easier to understand at first glance